Docs/Configuration.tex

```diff
@@ -7560,6 +7560,37 @@ \subsection{Properties}\label{uefiprops}
   could be the second 256 MB corrupted by the Intel HD 3000 or an area with faulty RAM.
   Refer to the \hyperref[uefirsvdprops]{ReservedMemory Properties} section below for details.
 
+\item
+  \texttt{Unload}\\
+  \textbf{Type}: \texttt{plist\ array}\\
+  \textbf{Failsafe}: Empty\\
+  \textbf{Description}: Unload specified firmware drivers.
+
+  To be filled with \texttt{plist\ string} entries containing
+  the names of firmware drivers to unload before loading the
+  \texttt{Drivers} section.
+  This setting is typically only required if a user-provided driver is a variant of
+  an existing system firmware driver, and if the new driver would detect itself
+  as partially loaded, or otherwise fail to operate correctly, if the old
+  driver is not unloaded first.
+
+  \textbf{Warning}: Unloading system firmware drivers is usually not required
+  and not recommended.
+
+  \emph{Note 1}: The list of driver which this option can attempt to unload
+  can be found in \texttt{SysReport/Drivers/DriverImageNames.txt}.
+  The relevant name is the driver component name. Drivers are only listed if they
+  implement \texttt{DriverBindingProtocol} and \texttt{LoadedImageProtocol},
```

Comment:
LoadedImage is installed by DxeCore, no?

Reply:
Huh? No. It's in MdeModulePkg/Core/Dxe for all loaded images, e.g.: https://github.com/acidanthera/audk/blob/master/MdeModulePkg/Core/Dxe/Image/Image.c#L841-L871

Reply:
That *is* DxeCore, so yes? :)

Reply:
We have an OC one, for our loader. It's not that one!

Reply:
I thought that was only used for booters? And even if not, point is that it’s an environment thing (DxeCore, OC, whatever), not a property of the driver itself.

Reply:
I'm going mad - I read OpenCore. Sorry. Okay. So then? Sorry! This was just double-checking?

Reply:
> I thought that was only used for booters? And even if not, point is that it’s an environment thing (DxeCore, OC, whatever), not a property of the driver itself.

The reason for checking for DXE Core gEfiLoadedImageProtocolGuid is because those are the handles which support gBS->UnloadImage, which is exactly what we want to use to unload these network stack drivers.

Reply:
No, you confused me with OcLoadedImage and I confused you back… everything is clear now! :)